bloom-housing/bloom
# fix: updated email and logic for single use code

This PR addresses #4349 and #4350


## Description

This PR updates the single use code logic so if a user requests a new code within the valid window, we send the same code rather than generating a new one. I verified with Yazeed and Morgan that the udpatedAt time should be refreshed to avoid the confusing experience where a user only has a few seconds to use a code. 

This PR also updates some single use code copy and related emails to show the requested ten minute window (will be updated via env when released) and the code in the subject line.

## How Can This Be Tested/Reviewed?

1. This can be tested by making a new account on the partners portal that uses mfa verification. Sign in and notice the updated email copy. Then before entering the mfa code, press the back button and sign in again. You'll notice a new email with the same verification code will be sent.
2. Enable single sign on for the partners portal and create a new account. Notice that all references to the single use code expiration now reference 10 minutes rather than 5. Sign in with single use code and then click resend before entering the code. Like above, this should show with the updated email copy but the same code.

**NOTE**
3. If you want to test the script runner, it requires a few steps. Setup the db using main branch, switch to this branch, and now run the script via the open api. Because of how our seeding works the single use code email translations exists both in the default and in the jurisdiction-specific translations. As a result, the email will still show 5 minutes because the script doesn't target the jurisdiction specific translations (and jurisdiction overrides default) but if you query the translations table you will see the updated duration under the row where jurisdiction id is null. I have verified on hba and doorway that their email translations exist as the default there too.

## Author Checklist:


## Review Process:

- Read and understand the issue
- Ensure the author has added QA notes
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Either (1) explicitly ask a clarifying question, (2) request changes, or (3) approve the PR, even if there are very small remaining changes, if you don't need to re-review after the updates
